cpp/src/cluster/detail/mst.cuh

```diff
@@ -299,12 +299,14 @@ void build_sorted_mst(
   int iters        = 1;
   int n_components = cuvs::sparse::neighbors::get_n_components(color, m, stream);
 
-  cudaPointerAttributes attr;
-  RAFT_CUDA_TRY(cudaPointerGetAttributes(&attr, X));
-  bool data_on_device = attr.type == cudaMemoryTypeDevice;
+  // TODO: Think about how to handle this for G+H. C-style malloc'd arrays can still lead to the
+  // device route being taken.
+  cudaPointerAttributes ptr_attrs;
```

Comment:
We have a function for this which was introduced more recently in cuda utils. We should also strive to use reusable logic for these types of things as much as possible so we can centralize these types of decisions and avoid having to make changes everywhere when logic and APIs need to be updated.

Reply:
Using raft::memory_type_from_pointer.